src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/MissingDocumentsRecordComponentTest.kt

```diff
@@ -50,59 +49,40 @@ class MissingDocumentsRecordComponentTest : ComponentTestBase() {
     assertThat(recallWithMissingDocumentsRecord.missingDocumentsRecords.first().emailFileName, equalTo(fileName))
   }
 
-  @Test
-  fun `create a MissingDocumentsRecord with temp form of request`() { // TODO PUD-1259: remove use of temp request and class when UI has transitioned
-    expectNoVirusesWillBeFound()
-    val recall = authenticatedClient.bookRecall(bookRecallRequest)
-    val missingDocsRecordReq = TempMissingDocumentsRecordRequest(
-      recall.recallId,
-      listOf(PART_A_RECALL_REPORT),
-      details = null,
-      base64EncodedDocumentContents,
-      fileName,
-      detail = "old detail prop"
-    )
-
-    val response = authenticatedClient.missingDocumentsRecord(missingDocsRecordReq, CREATED, MissingDocumentsRecordId::class.java)
-
-    assertThat(response, present())
-
-    val recallWithMissingDocumentsRecord = authenticatedClient.getRecall(recall.recallId)
-    assertThat(recall.missingDocumentsRecords, isEmpty)
-    assertThat(recallWithMissingDocumentsRecord.missingDocumentsRecords.size, equalTo(1))
-    assertThat(recallWithMissingDocumentsRecord.missingDocumentsRecords.first().version, equalTo(1))
-  }
-
   @Test
   fun `adding 2 MissingDocumentsRecords for the same recall, all missing documents will be returned on the recall`() {
     expectNoVirusesWillBeFound()
     val recall = authenticatedClient.bookRecall(bookRecallRequest)
-    val missingDocsRecordReq = TempMissingDocumentsRecordRequest(
+    val detailsOldest = "Some detail"
+    val missingDocsRecordReq = MissingDocumentsRecordRequest(
       recall.recallId,
       listOf(PART_A_RECALL_REPORT),
-      "Some detail",
+      detailsOldest,
       base64EncodedDocumentContents,
-      fileName,
-      null
+      fileName
     )
 
     authenticatedClient.missingDocumentsRecord(missingDocsRecordReq, CREATED, MissingDocumentsRecordId::class.java)
-    val response = authenticatedClient.missingDocumentsRecord(missingDocsRecordReq.copy(details = "Some details; some more detail"), CREATED, MissingDocumentsRecordId::class.java)
+    val detailsMostRecent = "Some details; some more detail"
+    val response = authenticatedClient.missingDocumentsRecord(missingDocsRecordReq.copy(details = detailsMostRecent), CREATED, MissingDocumentsRecordId::class.java)
 
     assertThat(response, present())
 
     val recallWithMissingDocumentsRecord = authenticatedClient.getRecall(recall.recallId)
     assertThat(recall.missingDocumentsRecords, isEmpty)
     assertThat(recallWithMissingDocumentsRecord.missingDocumentsRecords.size, equalTo(2))
     assertThat(recallWithMissingDocumentsRecord.missingDocumentsRecords[0].version, !equalTo(recallWithMissingDocumentsRecord.missingDocumentsRecords[1].version))
+    // TODO: following are unstable with the order of the returned records being non-deterministic when presumably we do want them to be
```

Comment:
This surprised me: and I didn't yet find why it might be this way.

Reply:
that's odd that its seemingly been working...

Reply:
I suspect it has not been working: the commented asserts are new here too.  Just we should be aware of this - which perhaps we weren't before? - and preferably fix it.